spec/unit/provider/oneview_server_profile_template_c7000_spec.rb

```diff
@@ -0,0 +1,150 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+require 'spec_helper'
+require_relative '../../support/fake_response'
+require_relative '../../shared_context'
+
+provider_class = Puppet::Type.type(:oneview_server_profile_template).provider(:c7000)
+api_version = login[:api_version] || 200
+resource_name = 'ServerProfileTemplate'
+resourcetype = if api_version == 200
+                 Object.const_get("OneviewSDK::API#{api_version}::#{resource_name}")
+               else
+                 Object.const_get("OneviewSDK::API#{api_version}::C7000::#{resource_name}")
+               end
+
+describe provider_class, unit: true do
+  include_context 'shared context'
+
+  context 'given the creation parameters' do
+    let(:resource) do
+      Puppet::Type.type(:oneview_server_profile_template).new(
+        name: 'spt',
+        ensure: 'present',
+        data:
+            {
+              'name'                  => 'SPT',
+              'enclosureGroupUri'     => '/rest/',
+              'serverHardwareTypeUri' => '/rest/',
+              'type'                  => 'ServerProfileTemplateV1'
+            },
+        provider: 'c7000'
+      )
+    end
+
+    let(:provider) { resource.provider }
+
+    let(:instance) { provider.class.instances.first }
+
+    it 'should be an instance of the provider c7000' do
+      expect(provider).to be_an_instance_of Puppet::Type.type(:oneview_server_profile_template).provider(:c7000)
+    end
+
+    it 'should run exists? and return the resource does not exist' do
+      allow(resourcetype).to receive(:find_by).and_return([])
+      expect(provider.exists?).to eq(false)
+    end
+
+    it 'should return the resource has been found' do
+      test = resourcetype.new(@client, resource['data'])
+      allow(resourcetype).to receive(:find_by).and_return([test])
+      provider.exists?
+      expect(provider.found).to be
+    end
+
+    it 'should return that the resource exists' do
+      test = resourcetype.new(@client, resource['data'])
+      allow(resourcetype).to receive(:find_by).and_return([test])
+      expect(provider.exists?).to eq(true)
+    end
+
+    it 'should be able to create the resource' do
+      allow(resourcetype).to receive(:find_by).and_return([])
+      allow_any_instance_of(resourcetype).to receive(:create).and_return(resourcetype.new(@client, resource['data']))
+      expect(provider.exists?).to eq(false)
+      expect(provider.create).to be
+    end
+
+    it 'should be able to create a server profile with default name using the template' do
+      test = resourcetype.new(@client, resource['data'])
+      allow(resourcetype).to receive(:find_by).with(anything, resource['data']).and_return([test])
```

Comment:
Should add this `allow` and other similar to a `before(:each)` method in order to avoid repetition